infra/Pos/Diffusion/Subscription/Common.hs

```diff
@@ -56,21 +61,37 @@ data SubscriptionTerminationReason =
 -- giving the reason. Notices will be logged before and after the subscription.
 subscribeTo
     :: forall m. (SubscriptionMode m)
-    => Timer -> SendActions m -> NodeId -> m SubscriptionTerminationReason
-subscribeTo keepAliveTimer sendActions peer = do
+    => Timer
+    -> TVar SubscriptionStatus
+    -> MVar Millisecond -- ^ Subscription duration.
+    -> SendActions m
+    -> NodeId
+    -> m SubscriptionTerminationReason
+subscribeTo keepAliveTimer subStatus subDuration sendActions peer = do
+    -- Change subscription status as we begin a new subscription
+    atomically $ writeTVar subStatus Subscribing
     logNotice $ msgSubscribingTo peer
+    subStarted <- liftIO getPOSIXTime
     outcome <- try $ withConnectionTo sendActions peer $ \_peerData -> NE.fromList
         -- Sort conversations in descending order based on their version so that
         -- the highest available version of the conversation is picked.
         [ Conversation convMsgSubscribe
         , Conversation convMsgSubscribe1
         ]
+    -- Change subscription state
+    atomically $ writeTVar subStatus NotSubscribed
```

Comment:
This is a bit trickier: if there's some other subscription active, don't go to NotSubscribed.